### ISSUE TITLE
FPDF_RenderPageBitmap" failed: JsValue(RuntimeError: call_indirect to a null table entry (evaluating 'asm[name].apply(null,arguments)

### ISSUE THREAD

**Issue by reyjexter** [ajrcarey/pdfium-render#125]
Hello,

We ran into this issue with some documents when using Safari (both desktop and iphone). The web stack trace is as follow.

<img width="1360" alt="image" src="https://github.com/ajrcarey/pdfium-render/assets/669590/240770c7-dee7-4d87-bddb-3f8ff640a30f">

This doesn't happen on all documents and can occur only on certain documents. Also this only appears to happen on Safari but not Chrome and Brave browser.

It appears that the issue occurs somewhere when retrieving the image data on this function:

https://github.com/reyjexter/pdfium-render-wasm/blob/master/src/lib.rs#L89

_Note: I haven't attached the pdf on which this error occurs as I'm still asking for an approval due to document may content sensitive data but hopefully if this error will help._

Thanks

EDIT: Made some edits on extra text I pasted.

**Comment by ajrcarey** [comment]
Hi @reyjexter , the error is thrown here: https://github.com/ajrcarey/pdfium-render/blob/master/src/wasm.rs#L401

Please switch on debugging output in your initialization of pdfium-render, as shown at https://github.com/ajrcarey/pdfium-render/blob/master/examples/index.html#L65, and provide those logs. Additionally, when filing any bug report for any project, please provide the versions of all relevant software; in this case, your Safari version and the source and version of your WASM Pdfium build.

**Comment by reyjexter** [comment]
Here's the software that we use:

MacOS - 14.1.1 (23B81) Sonoma
Version 17.1 (19616.2.9.11.7)
WASM Pdfium: V5407
pdfium-render: master branch
Pdf size: 500 KB

I also attached the stack trace that is  shown on log.

[stacktrace.txt](https://github.com/ajrcarey/pdfium-render/files/13517561/stacktrace.txt)

**Comment by reyjexter** [comment]
We were able to replicate the issue with a test document and the issue appears to be happening if you create a PDF that included rounded corners. Here's the document which is a Canva generated document:

https://github.com/reyjexter/pdfium-render-wasm/blob/master/www/panic.pdf

**Comment by ajrcarey** [comment]
Possibly related to https://bugs.webkit.org/show_bug.cgi?id=263913

**Comment by reyjexter** [comment]
It looks like this is indeed a webkit issue. From our tests, this is already fixed on the new Safari 17.2 Beta4 and this change log probably fixed that:

https://developer.apple.com/documentation/safari-release-notes/safari-17_2-release-notes#Web-Assembly

Prior to testing on the 17.2 Beta4 Safari, we also tried compiling on Mac and Linux from 1.74 down to 1.61 and the issue didn't got fixed.

Thanks for providing the details especially the link to webkit issue as it was very useful to track this down.
